modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -2592,12 +2592,14 @@ private void updateScrollBarsAndCells(boolean recreate) {
         // thumb can be appropriately sized
         if (recreate && (lengthBar.isVisible() || Properties.IS_TOUCH_SUPPORTED)) {
             final int cellCount = getCellCount();
-            int numCellsVisibleOnScreen = 0;
+            float numCellsVisibleOnScreen = 0;
```

Comment:
Can you explain why this needs to be in floating point?

Presuming that it does need to be in floating-point, I recommend making this a double to match `lengthToAdd`.